### PR TITLE
[fix] UT MathFunctionTest.round_test fix

### PR DESCRIPTION
**Pull request by cambyzju**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:
Function round support two format round(double) and round(double, int), the argument is variadic.
But FunctionBinaryArithmetic not support variadic argument now, make get_function for round(double, int) failed.

reproduce steps:
1. set enable_vectorized_engine=true;
2. try to call round(double, int);
```
> select round(10.12345,2);
ERROR 1105 (HY000): errCode = 2, detailMessage = Function round is not implemented
```

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
4. Has unit tests been added: (Yes/No/No Need)
5. Has document been added or modified: (Yes/No/No Need)
6. Does it need to update dependencies: (Yes/No)
7. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_binary_arithmetic.h [hunk 1]
```diff
@@ -756,6 +756,16 @@ class FunctionBinaryArithmetic : public IFunction {
 
     size_t get_number_of_arguments() const override { return 2; }
 
+    DataTypes get_variadic_argument_types_impl() const override {
+        using OpImpl = Operation<typename OpTraits::T, typename OpTraits::T>;
+        constexpr bool has_variadic_argument =
```
Comment (BiteTheDDDDt) [inline be/src/vec/functions/function_binary_arithmetic.h:761]: Maybe we can move `has_variadic_argument` to OperationTraits?

**Comment by HappenLee** [comment]
Add a regression test for the case. thank you ~~

**Comment by xinyiZzz** [comment]
```
==47967==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x55d6edb1fc2f bp 0x7ffeb746cd50 sp 0x7ffeb74698d0 T0)
==47967==The signal is caused by a READ memory access.
==47967==Hint: address points to the zero page.
    #0 0x55d6edb1fc2f in void doris::vectorized::check_function<doris::vectorized::DataTypeNumber<double>, true>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<std::any, std::allocator<std::any> > const&, std::vector<std::pair<std::vector<std::any, std::allocator<std::any> >, std::any>, std::allocator<std::pair<std::vector<std::any, std::allocator<std::any> >, std::any> > > const&) /home/disk3/zxy/baidu/bdg/doris/core/be/test/vec/function/function_test_util.h:162
    #1 0x55d6edafbf7b in doris::vectorized::MathFunctionTest_round_test_Test::TestBody() /home/disk3/zxy/baidu/bdg/doris/core/be/test/vec/function/function_math_test.cpp:382
    #2 0x55d6f7baf7dc in void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d9c7dc)
    #3 0x55d6f7ba9c18 in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d96c18)
    #4 0x55d6f7b812fb in testing::Test::Run() (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d6e2fb)
    #5 0x55d6f7b81d1f in testing::TestInfo::Run() (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d6ed1f)
    #6 0x55d6f7b825df in testing::TestSuite::Run() (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d6f5df)
    #7 0x55d6f7b91c9f in testing::internal::UnitTestImpl::RunAllTests() (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d7ec9f)
    #8 0x55d6f7bb058d in bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d9d58d)
    #9 0x55d6f7baaad2 in bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*) (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d97ad2)
    #10 0x55d6f7b90494 in testing::UnitTest::Run() (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x22d7d494)
    #11 0x55d6ed423126 in RUN_ALL_TESTS() /home/disk3/zxy/baidu/bdg/doris/core/thirdparty/installed/include/gtest/gtest.h:2490
    #12 0x55d6ed41f6f2 in main /home/disk3/zxy/baidu/bdg/doris/core/be/test/testutil/run_all_tests.cpp:43
    #13 0x7fb028695cdc in __libc_start_main (/lib64/libc.so.6+0x318ae1ecdc)
    #14 0x55d6ebd35579 in _start (/home/disk3/zxy/baidu/bdg/doris/core/be/ut_build_ASAN/test/doris_be_test+0x16f22579)
```

Is this the error?
I got this error when I run be UT, apply your patch to solve it.

**Comment by cambyzju** [comment]
> function_math_test.cpp

yes.

/home/disk3/zxy/baidu/bdg/doris/core/be/test/vec/function/function_math_test.cpp:382
failed because of failed to find function round(double, int).

**Comment by cambyzju** [comment]
> Add a regression test for the case. thank you ~~

done

**Comment by BiteTheDDDDt** [comment]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**